### ISSUE TITLE
AEPAssurance | non-modular header

### ISSUE THREAD

**Issue by vijayajay17** [Adobe-Marketing-Cloud/acp-sdks#585]
Hi Team,

We are migrating from 4x to new Exp.. platform SDK, when ever we are trying to import AEPAssurance header we are getting the following error

"include of non-modular header inside framework module 'OurSDKModule'"

we are importing the header in the .h file
#import "AEPAssurance.h"

Note: We had a similar issue for AEPMobileServices framework previously,
https://github.com/Adobe-Marketing-Cloud/acp-sdks/issues/582#issuecomment-857816936

Thanks
Vijay

**Comment by PravinPK** [comment]
Hey @vijayajay17, 
Can u try this AEPAssurance build from the [master branch](https://github.com/Adobe-Marketing-Cloud/acp-sdks/tree/master/iOS/AEPAssurance/AEPAssurance.xcframework) 
Also, may I know from where did u previously download the xcframework from? 

Thanks,
Pravin

**Comment by vijayajay17** [comment]
Hi @PravinPK,

Thanks for the quick response, after using the framework from master branch, we are getting the below error

```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_ACPExtensionListener", referenced from:
      _OBJC_CLASS_$_AEPAssuranceListenerAssuranceRequestContent in libAEPAssurance_iOS.a(AEPAssuranceListenerAssuranceRequestContent.o)
      _OBJC_CLASS_$_AEPAssuranceListenerHubPlacesRequests in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesRequests.o)
      _OBJC_CLASS_$_AEPAssuranceListenerHubPlacesResponses in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesResponses.o)
      _OBJC_CLASS_$_AEPAssuranceListenerHubWildcard in libAEPAssurance_iOS.a(AEPAssuranceListenerHubWildcard.o)
  "_OBJC_METACLASS_$_ACPExtensionListener", referenced from:
      _OBJC_METACLASS_$_AEPAssuranceListenerAssuranceRequestContent in libAEPAssurance_iOS.a(AEPAssuranceListenerAssuranceRequestContent.o)
      _OBJC_METACLASS_$_AEPAssuranceListenerHubPlacesRequests in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesRequests.o)
      _OBJC_METACLASS_$_AEPAssuranceListenerHubPlacesResponses in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesResponses.o)
      _OBJC_METACLASS_$_AEPAssuranceListenerHubWildcard in libAEPAssurance_iOS.a(AEPAssuranceListenerHubWildcard.o)
  "_OBJC_METACLASS_$_ACPExtension", referenced from:
      _OBJC_METACLASS_$_AEPAssuranceInternal in libAEPAssurance_iOS.a(AEPAssuranceInternal.o)
  "_OBJC_CLASS_$_ACPExtension", referenced from:
      _OBJC_CLASS_$_AEPAssuranceInternal in libAEPAssurance_iOS.a(AEPAssuranceInternal.o)
  "_OBJC_CLASS_$_ACPExtensionEvent", referenced from:
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssurance.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssurancePluginFakeEventGenerator.o)
  "_OBJC_CLASS_$_ACPCore", referenced from:
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssurance.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceInternal.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceEvent.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceListenerAssuranceRequestContent.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesRequests.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceListenerHubPlacesResponses.o)
      objc-class-ref in libAEPAssurance_iOS.a(AEPAssuranceListenerHubWildcard.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I have uploaded a sample app, that might help u to understand more on the issue.
https://drive.google.com/file/d/1epQY2ER6RXXPUQFJLDzW6X25v6xlsGqF/view?usp=sharing

Also we downloaded the framework from the below link
https://github.com/Adobe-Marketing-Cloud/acp-sdks/releases

Thanks
Vijay

**Comment by PravinPK** [comment]
@vijayajay17, thank you for the sample app. We will get back to you soon.

Best,
Pravin

**Comment by PravinPK** [comment]
Hi @vijayajay17,

The problem that you see is because AEPAssurance SDK requires installation of an additional framework to be compatible with AEPCore.
Here is the link to that additional framework for you to [download](https://github.com/PravinPK/assurance_forum_fix/tree/main/Additional%20framework) and include in your application.
Here is the [sample app](https://github.com/PravinPK/assurance_forum_fix/tree/main/workingSampleApp) that includes this framework and builds alright.

**To Note:** We are currently working on an update to Assurance SDK that will not require this additional compatibility framework. It is expected to be released sometime by the end of next week. I will personally get in touch with you about this release and the updated install instruction.


**Release builds**
For AEPAssurance and AEPMobileServices you can download the latest release builds from [here](https://github.com/Adobe-Marketing-Cloud/acp-sdks/tree/master/iOS).

The releases [here](https://github.com/Adobe-Marketing-Cloud/acp-sdks/releases) does not currently include the module map support. I will work with our team to update them.

Let me know if have any questions.

Thanks,
Pravin

**Comment by PravinPK** [comment]
Hi @vijayajay17 ,

Our team has released a new Assurance SDK (version 3.0.0). That does not require the additional ACPCore Compatibility framework to work with AEPCore.

**Installation**
The Assurance project has been open-sourced. To obtain a xcframework binary, you can follow this [instruction](https://github.com/adobe/aepsdk-assurance-ios#binaries).
Furthermore, You can now remove the ACPCore xcframework from your project that was proposed to you in my earlier comment. 


**Migration** 
There are a few minor changes on the API's in our new AEPAssurance 3.x swift library. Follow this guide for [migration](https://aep-sdks.gitbook.io/docs/foundation-extensions/adobe-experience-platform-assurance/migration).
Here is an [example](https://github.com/adobe/aepsdk-sample-app-ios/pull/41 ).   

Closing this ticket. Please feel free to reopen if you have any questions.
